geneate_dynein_conformation_graphs/englishORspanish.py

```diff
@@ -0,0 +1,199 @@
+    angles = [] # Initialize a list to store the angles between consecutive direction vectors
+
+    # Loop through the data to create direction vectors for each segment
+    for i in range(num_segments):
+        # Extract the segment from the data
+        segment = data[i * segment_length:(i + 1) * segment_length]
+        
+        # Calculate the mean of the segment
+        segment_mean = segment.mean(axis=0)
+
+        # Perform Singular Value Decomposition (SVD) on the mean-centered segment
+        uu, dd, vv = np.linalg.svd(segment - segment_mean)
+        
+        # The first right singular vector (vv[0]) represents the direction of the segment
+        direction_vectors.append(vv[0])
+
+    # Loop through the direction vectors to calculate angles between consecutive vectors    
+    for i in range(len(direction_vectors) - 1):
+        vec1 = direction_vectors[i]
+        vec2 = direction_vectors[i + 1]
+        
+        # Calculate the cosine of the angle between the two vectors   
+        # Using np.clip to prevent numerical issues by ensuring the cosine value is within [-1, 1]     
+        angle = np.arccos(np.clip(np.dot(vec1, vec2) / (np.linalg.norm(vec1) * np.linalg.norm(vec2)), -1.0, 1.0))
+        
+        # Calculate the angle in degrees
+        angle_degrees = np.degrees(angle)
+
+        # Filter out angles that are greater than the angle_threshold
+        if angle_degrees <= angle_threshold: 
+            angles.append(angle_degrees)
+    
+    return angles
+
+# Function to process the angle data 
+def calculate_fit(data_lists, foldername):
+    # Adjustable parameters, after optimizing, these were the best settings
+    angle_threshold = 6  # When the angle is greater than x degrees, it is considered a bend
+    bend_threshold = 100 # Everything greater than x degrees, will be removed because junkdata
+    num_segments = 10    # How many segments per doublet
+
+    # Initialize empty lists
+    all_angles = [] 
+    bent_lists = [] 
+    bending_amounts = []
+    angle_differences = []
+
+    # Loop through each doublet of data points
+    for index, data in enumerate(data_lists):
+        # Calculate angles for the current doublet
+        angles = calculate_angles(data, num_segments, bend_threshold)
+        all_angles.append(angles)
+        
+        # Sum of absolute angles as bending metric
+        bending_amount = np.sum(np.abs(angles))  
+        bending_amounts.append(bending_amount)
+
+        # Filter all angles greater than the angle_threshold
+        if any(angle > angle_threshold for angle in angles):
+            bent_lists.append(index+1)
+
+    # Print results
+    if print_results:
+        for i, (angles, bending_amount) in enumerate(zip(all_angles, bending_amounts)):
+            print(f"Angles for list {i+1} (in degrees): {angles}")
+            print(f"Bending amount for list {i+1}: {bending_amount}")
```

Comment:
to start index counting at one, it is cleaner to use `start=1` in `enumerate
```suggestion
        for i, (angles, bending_amount) in enumerate(zip(all_angles, bending_amounts), start=1):
            print(f"Angles for list {i} (in degrees): {angles}")
            print(f"Bending amount for list {i}: {bending_amount}")
```